dev-tools/mage/integtest.go

```diff
@@ -242,19 +396,77 @@ func IsInIntegTestEnv() bool {
 	return found
 }
 
-func haveIntegTestEnvRequirements() error {
-	if err := HaveDockerCompose(); err != nil {
-		return err
+// IntegrationEnv represents the types of integrations environment the dir needs.
+type IntegrationEnv struct {
+	types map[string]string
+}
+
+// NewIntegrationEnvFromDir gets the integration environment for the dir.
+func NewIntegrationEnvFromDir(dir string) *IntegrationEnv {
+	cwd, err := os.Getwd()
+	if err != nil {
+		panic(err)
 	}
-	if err := HaveDocker(); err != nil {
-		return err
+	types := map[string]string{}
+	dockerFile := filepath.Join(cwd, dir, "docker-compose.yml")
+	if _, err := os.Stat(dockerFile); !os.IsNotExist(err) {
+		types[TestingEnvDocker] = dockerFile
+	}
+	kubeFile := filepath.Join(cwd, dir, "kubernetes.yml")
+	if _, err := os.Stat(kubeFile); !os.IsNotExist(err) {
+		types[TestingEnvKubernetes] = kubeFile
+	}
+	return &IntegrationEnv{types}
```

Comment:
@jsoriano do you see a good fit for using terraform here too? :innocent:

Reply:
Yes, this will be definitely an option if we move terraform scenarios to modules :slightly_smiling_face:

Reply:
Yes, I think this would be a great place to connect with terraform, as I would bet that most `*.tf` will be module specific.

Could even been combined together:

1. Deploy GKE with Terraform
2. Run kubernetes integration tests in GKE.

That was the exact reason I put a lot of work in running the tests inside of a Pod inside of Kubernetes to allow integration tests with cloud kubernetes deployments.

Reply:
Just for reference and since maybe we would like to test on Openshift too in the future: Openshift installation on GCP is achieved with their installer which uses `terafform` underneath (see https://github.com/elastic/beats/issues/17516#issuecomment-617038764).